spock-core/src/main/java/spock/mock/MockMakerId.java

```diff
@@ -0,0 +1,16 @@
+package spock.mock;
+
+import org.spockframework.mock.runtime.IMockMaker;
+
+/**
+ * Provides constants for known {@link IMockMaker} implementations.
+ */
+public final class MockMakerId {
+  private MockMakerId() {
+
+  }
+
+  public static final String byteBuddy = "byte-buddy-mock-maker";
+  public static final String cgLib = "cglib-mock-maker";
+  public static final String dynamicProxy = "dynamic-proxy-mock-maker";
```

Comment:
🤔 maybe we should use `javaProxy`/`jdkProxy` as public name.

Reply:
I have changed it now to `java-proxy-mock-maker`.